File: MedicationManagement/Models/IoTDevice.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MedicationManagement.Models
{
    // IoTDevice class
    public class IoTDevice
    {
        // Device property
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int DeviceID { get; set; }
        [StringLength(50)]
        public string Location { get; set; }
        [StringLength(50)]
        public string Type { get; set; }
        public string Parameters { get; set; }
        public bool IsActive { get; set; } = false;
        public float MinTemperature { get; set; }
        public float MaxTemperature { get; set; }
        public float MinHumidity { get; set; }
        public float MaxHumidity { get; set; }
    }
}

File: Mobile/MedicationManagement/app/src/main/java/com/example/medicationmanagement/model/IoTDevice.kt
package com.example.medicationmanagement.model

data class IoTDevice(
    val deviceID: Int,
    val location: String,
    val type: String,
    val parameters: String,
    val isActive: Boolean,
    val minTemperature: Double,
    val maxTemperature: Double,
    val minHumidity: Double,
    val maxHumidity: Double
)
